Tests/SwiftDocCTests/Infrastructure/DocumentationContext/DocumentationContextTests.swift

```diff
@@ -3908,6 +3908,25 @@ let expected = """
             "Expectedly returned non-nil value for non-symbol content."
         )
     }
+
+    func testDiagnosticLocations() throws {
+        // The ObjCFrameworkWithInvalidLink.docc test bundle contains symbol
+        // graphs for both Obj-C and Swift, built after setting:
+        //   "Build Multi-Language Documentation for Objective-C Only Targets" = true.
+        // One doc comment in the Obj-C header file contains an invalid doc
+        // link on line 24, columns 56-63:
+        // "Log a hello world message. This line contains an ``invalid`` link."
+        let (_, _, context) = try testBundleAndContext(copying: "ObjCFrameworkWithInvalidLink")
+        let problems = context.problems
+        XCTAssertEqual(1, problems.count)
+        let problem = try XCTUnwrap(problems.first)
+        let basename = try XCTUnwrap(problem.diagnostic.source?.lastPathComponent)
+        XCTAssertEqual("HelloWorldFramework.h", basename)
+        let start = Markdown.SourceLocation(line: 24, column: 56, source: nil)
```

Comment:
Question: should these source locations have a `source` value? I genuinely don't remember when that's set and when it's not.

Reply:
It looks like source is the path/URL of the markdown file, which are parsed by the swift-markdown package. However, in this test and code path, the markdown comes from a doc comment from a Swift code file... in this test one of the symbol graph files under `Test Bundles/ObjCFrameworkWithInvalidLink.docc`. When DocC parses markdown from doc comments, it looks like it leaves the source URL as the default value nil, here in the call to the `Document` initializer:

```
let documentOptions: ParseOptions = [.parseBlockDirectives, .parseSymbolLinks, .parseMinimalDoxygen]
let docCommentMarkup = Document(parsing: docCommentString, options: documentOptions)
```
(from: https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Model/DocumentationNode.swift#L442)

Anyway, this test is about checking the line and column values returned by the call to:

```
let offset = symbol.offsetAdjustedForInterfaceLanguage()
```

...which I added in this PR and immediately follows the first snippet above on my branch. So the value of source isn't relevant to this test anyway.